central/image/service/service_impl.go

```diff
@@ -203,14 +208,48 @@ func internalScanRespFromImage(img *storage.Image) *v1.ScanImageInternalResponse
 	}
 }
 
-func (s *serviceImpl) saveImage(img *storage.Image) error {
+func (s *serviceImpl) saveImage(ctx context.Context, img *storage.Image) error {
+	if features.ScannerV4Enabled.Enabled() && !scannedByScannerV4(img) && s.scannedByClairify(img) {
```

Comment:
nit: if you want to create a variable to capture `scannerByScannerV4` so you don't have to call it twice

Reply:
The image passed to `scannedByScannerV4` is different in the two use cases FWIW